### ISSUE TITLE
Avoid re-downloading multiple identical LLVMs from CI

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#76587]
Any LLVM submodule update landing means that we can switch rust-dev downloads added by #76349 to look for the LLVM submodule update rather than latest bors commit. That'll avoid spuriously redownloading essentially identical tarballs whenever someone rebases.

This is technically a QoL issue so does not block that PR, but is a rather major improvement since rebasing is quite common and the download is relatively hefty at ~20 MB.

**Comment by RalfJung** [comment]
Does "rebasing" here include "pulling" (to get latest master before starting a new PR)? I am confused by the emphasis on rebasing.

**Comment by Mark-Simulacrum** [comment]
Yes, any transition of "base" commit will cause a redownload. The option is currently off by default, though. I think most people would rather a download than a potential LLVM rebuild, and in any case this should get fixed in the next week or two I expect.

**Comment by Mark-Simulacrum** [comment]
I believe this has since been fixed, so closing.
